beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -178,57 +198,57 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
 
         // Spec: Full nodes SHOULD provide the best derivable LightClientUpdate (according to is_better_update)
         // for each sync committee period
-        let prev_light_client_update = match &self.latest_light_client_update.read().clone() {
-            Some(prev_light_client_update) => Some(prev_light_client_update.clone()),
-            None => self.get_light_client_update(&store, sync_period, chain_spec)?,
-        };
+        let prev_light_client_update =
+            self.get_light_client_update(&store, sync_period, chain_spec)?;
 
         let should_persist_light_client_update =
             if let Some(prev_light_client_update) = prev_light_client_update {
-                let prev_sync_period = prev_light_client_update
-                    .signature_slot()
-                    .epoch(T::EthSpec::slots_per_epoch())
-                    .sync_committee_period(chain_spec)?;
-
-                if sync_period != prev_sync_period {
-                    true
-                } else {
-                    prev_light_client_update
-                        .is_better_light_client_update(&new_light_client_update, chain_spec)?
-                }
+                prev_light_client_update
+                    .is_better_light_client_update(&new_light_client_update, chain_spec)?
             } else {
                 true
             };
 
         if should_persist_light_client_update {
-            self.store_light_client_update(&store, sync_period, &new_light_client_update)?;
+            store.store_light_client_update(sync_period, &new_light_client_update)?;
+            *self.latest_light_client_update.write() = Some(new_light_client_update);
         }
 
         Ok(())
     }
 
-    fn store_light_client_update(
+    fn store_current_sync_committee(
         &self,
         store: &BeaconStore<T>,
+        cached_parts: &LightClientCachedData<T::EthSpec>,
         sync_committee_period: u64,
-        light_client_update: &LightClientUpdate<T::EthSpec>,
+        finalized_period: u64,
     ) -> Result<(), BeaconChainError> {
-        let column = DBColumn::LightClientUpdate;
-
-        store.hot_db.put_bytes(
-            column.into(),
-            &sync_committee_period.to_le_bytes(),
-            &light_client_update.as_ssz_bytes(),
-        )?;
+        if let Some(latest_sync_committee) =
+            self.latest_written_current_sync_committee.read().clone()
+        {
+            if latest_sync_committee == cached_parts.current_sync_committee {
+                return Ok(());
+            }
+        };
 
-        *self.latest_light_client_update.write() = Some(light_client_update.clone());
+        if finalized_period >= sync_committee_period - 1 {
```

Comment:
doesn't this overflow for `sync_committee_period == 0`?

could switch it to:

```
finalized_period + 1 >= sync_committee_period
```

Reply:
[82a57b5](https://github.com/sigp/lighthouse/pull/5915/commits/82a57b591be27569b0380e8ef931e23b48ca204c)